tunnel: pass our stderr to the pppd slave

lkundrak opened the pull request:
While forkpty() does a nice job of convenently setting up a pty for the
pppd, it goes a bit too far for our purposes by also sending stderr to
the pseudoterminal.

This causes confusion in the event pppd (or its plugin, or us, for that
matter) try to log anything to stderr. Let's restore the parent's stderr
after he fork.

DimitriPapadopoulos commented:
I had been trying to fix this for some time now. Thank you!

lkundrak commented:
I think this is the issue that was causing trouble with the NM plugin: https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn/issues/9

lkundrak commented:
One way to test this is by passing `--pppd-log /dev/stderr`

ksyz commented:
Hi, 
just want to confirm, that with [openfortivpn-1.9.0-1.fc31.lr1.src.rpm](https://koji.fedoraproject.org/koji/taskinfo?taskID=33651854) I was able to successfully configure and connect Fortinet VPN via NetworkManager. Finaly.

adrienverge commented:
Thanks @lkundrak!

DimitriPapadopoulos commented:
@lkundrak I can remove the `TODO` from your own branch myself - if you're OK with that.

lkundrak commented:
> @lkundrak I can remove the `TODO` from your own branch myself - if you're OK with that.

Of course it would've been okay. I've removed this now though, and also added another patch, based on the feedback in the NM plugin ticket. Apparently ppp loves to log to stdout by default, because why not...

lkundrak commented:
Anyone minds if I merge this?

adrienverge commented:
Good to me :+1:

mrbaseman commented:
this went into the [1.10.0 Release](https://github.com/adrienverge/openfortivpn/releases/tag/v1.10.0).
